### PR TITLE
feat: use timestamp in changeset file name [PHX-2937]

### PR DESCRIPTION
**Pull request by mayakarabula**
The purpose of this PR is to make changeset name unique and thus not overwrite the changeset every time a create command is run, the new format is `changeset-[DATE]-[SPACE]-[SOURCE]-[TARGET].json`

It is however possible to pass an `output-file` flag and set it to any file

This change in logic prompts a change of the apply command `file` flag to be required

### CODE REVIEW

File: src/commands/apply/index.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default class Apply extends Command {
   static flags = {
     space: Flags.string({ default: undefined, description: 'Space id', required: true }),
     environment: Flags.string({ default: undefined, description: 'Target environment id', required: true }),
-    file: Flags.string({ description: 'File path to changeset file', required: false, default: 'changeset.json' }),
+    file: Flags.string({ description: 'File path to changeset file', required: true, default: undefined }),
```
Comment (marcolink) [inline src/commands/apply/index.ts:49]: Ahouldn't this be optional, and if not provided fall back to the new schema 🤔

Comment (ruderngespra) [reply src/commands/apply/index.ts:49]: But this is on the apply and we can't fall back on the new schema as we don't know the specifics of the changeset file (e.g. timestamp, env names etc.)

File: src/commands/create/index.ts [hunk 2]
```diff
@@ -182,7 +183,9 @@ export default class Create extends Command {
       num_target_entries: context.targetData.entries.ids.length,
     })
 
-    this.changesetFilePath = path.join(process.cwd(), 'changeset.json')
+    this.changesetFilePath =
+      flags['output-file'] ||
+      path.join(process.cwd(), `changeset-${Date.now()}-${flags.space}-${flags.source}-${flags.target}.json`)
```
Comment (ruderngespra) [inline src/commands/create/index.ts:188]: is this the same structure we use in `contentful-export` and `contentful-import`?

**Review by ruderngespra — CHANGES_REQUESTED** [review]
Looks very nice, just wondering, how did we align on the format and is it the same that we use in our other tooling?

File: src/commands/apply/index.ts (hunk 1 above)
Comment (marcolink) [reply src/commands/apply/index.ts:49]: I missed the fact that this is `apply` 🙈

File: src/commands/create/index.ts (hunk 2 above)
Comment (mayakarabula) [reply src/commands/create/index.ts:188]: yes

**Review by ruderngespra — APPROVED** [review]

**Pull request merged**